pkg/provider/azure_test.go

```diff
@@ -1620,6 +1620,13 @@ func getInternalTestService(identifier string, requestedPorts ...int32) v1.Servi
 	svc.Annotations[consts.ServiceAnnotationLoadBalancerInternal] = consts.TrueAnnotationValue
 	return svc
 }
+func getInternalTestServiceWithAnnotation(identifier string, annotations map[string]string, requestedPorts ...int32) v1.Service {
```

Comment:
How about implementing factory logic in getInternalTestServiceWithAnnotation and refactor getInternalTestService instead? Other functions may reuse logics in annotations without duplicating service generation logic.

Reply:
yes, that should help for new test suites